src/repository/views.py

```diff
@@ -1537,16 +1537,24 @@ def repository_wizard(request, short_name=None, step='1'):
                     )
                 )
 
-            # Bump the step by 1
-            kwargs = {'step': int(step) + 1}
-            if updated_repository:
-                kwargs['short_name'] = updated_repository.short_name
-            return redirect(
-                reverse(
-                    'repository_wizard_with_id',
-                    kwargs=kwargs,
+            kwargs = {'short_name': updated_repository.short_name}
+
+            if 'next' in request.POST:
+                kwargs['step'] = str(int(step) + 1)
+                return redirect(
+                    reverse(
+                        'repository_wizard_with_id',
+                        kwargs=kwargs,
+                    )
+                )
+            else:
+                kwargs['step'] = step
+                return redirect(
+                    reverse(
+                        'repository_wizard_with_id',
+                        kwargs=kwargs,
+                    )
```

Comment:
minor nitpick: Seems like inly the `kwargs["step"]` definition should live in the if block, the URL is still the same in both cases